### ISSUE TITLE
Add HTTP -> HTTPS redirection on App Gateway config

### ISSUE THREAD

**Issue by hosungsmsft** [Azure/Moodle#102]
#98 added the Azure App Gateway support (fixing #31), but it lacks the http://siteurl -> https://siteurl redirection, which should be very easy with an Azure App Gateway example ([doc](https://docs.microsoft.com/en-us/azure/application-gateway/tutorial-http-redirect-cli)). Community contribution would be great on things like this.

**Comment by ChrisLopezUK** [comment]
Hi @hosungsmsft. I think I have nailed this issue.

In addition I ammended the templates to allow for a conditional AppGW V2 deployment (which is a huge improvement over V1).

Not sure how to contribute. I took a clone of the repo and made a separate branch. Not sure what to do now. Any tips?

**Comment by hosungs** [comment]
> Hi @hosungsmsft. I think I have nailed this issue.
> 
> In addition I ammended the templates to allow for a conditional AppGW V2 deployment (which is a huge improvement over V1).
> 
> Not sure how to contribute. I took a clone of the repo and made a separate branch. Not sure what to do now. Any tips?

Sorry @ChrisLopezUK , I no longer work at Microsoft on this project. @rgardler might be able to point you to the right direction. Thanks.

**Comment by rgardler** [comment]
Hi @ChrisLopezUK,

If you cloned the repo and made your changes there then you already did 95% of the work - thank you for that. We'd love to merge these changes in. The overall process is in the [Understanding the GitHub flow](https://guides.github.com/introduction/flow/).

The last mile is to issue a pull request against this repo. It sounds like you cloned this repo and made the changes locally, you have not already pushed the changes to a fork on Github. Assuming this is the case here is what you need to do (before following these steps assume I have made at least one error and thus copy your local files to a backup directory before starting).

1. Create a fork of this repo (short version, click the Fork button top right, longer version in [docs](https://help.github.com/en/github/getting-started-with-github/fork-a-repo)

2. Make your fork the origin for your local clone (short version below, or [docs](https://help.github.com/en/github/using-git/changing-a-remotes-url)

```
git remote set-url origin YOUR_REPO
```

3. Commit the changes locally. Please ensure only changes relevant to this issue are included.

```
git add CHANGED_FILES
git commit -m "Adding HTTPS redirection on App Gateway"
```

4. Push your changes to your fork ([docs](https://help.github.com/en/github/using-git/pushing-commits-to-a-remote-repository))

```
git push
```

5. Create a pull request against this repo and referencing this issue in the comment with "Fixes #102" (this will tag the PR with this issue and vice-versa) see https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork

From there we will review the changes, possibly with a few requests for improvement and eventually we should be able to merge.

Thank you for your contribution, we truly appreciate it.
